### DISCUSSION TITLE
How to save file when I press `Esc` (enter normal mode)

### DISCUSSION THREAD

**Discussion by CyrusYip** [JetBrains/ideavim#629] [Q&A]
Basically, I want this plugin (https://github.com/907th/vim-auto-save).

**Comment by AlexPl292** [comment]
Hi there. I guess, this can be achieved by `map <esc> <esc>:w<cr>` mapping or something like that?
Actually, this plugin says that it's inspired by RubyMine feature. RubyMine IDE is one of the IntelliJ IDEs, so if you use IdeaVim plugin, you already have this feature because of IntelliJ itself.

**Reply by CyrusYip** [reply]
I installed IdeaVim on Webstorm. I enable prettier on save. I don't have this feature. Do I need to toggle it somewhere?


In the video, you can see the file was not immediately saved when I press Esc.

https://user-images.githubusercontent.com/60951091/234803310-63077073-ce0b-428e-a0f3-fc46b9c7308b.mp4

.ideavimrc: https://github.com/CyrusYip/dotfiles/blob/878e2c77df46a1757b9175759a53758fc10b5b42/dot_ideavimrc

autosave settings: 
![image](https://user-images.githubusercontent.com/60951091/234804917-e13485a9-387f-4357-b438-8d2b9aef9662.png)

**Reply by AlexPl292** [reply]
Oh, yes, IJ doesn't support automatic actions. I've created a request for this plugin: https://youtrack.jetbrains.com/issue/VIM-2935/Implement-vim-autosave-plugin
